### PR TITLE
chore: Fix logits dtype in assert

### PR DESCRIPTION
**Pull request by achartier**
Remove extra methods in trtGptModelInflightBatching.h. The methods were moved out of that class during a previous refactoring, but the definitions have been left behind.

### CODE REVIEW

**Comment by niukuo** [comment]
[PR_Github #326](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/326/) [ run ] triggered by Bot

File: cpp/tensorrt_llm/batch_manager/trtGptModelInflightBatching.h
Comment (symphonylyh) [inline cpp/tensorrt_llm/batch_manager/trtGptModelInflightBatching.h:1]: for removes in this file, are they due to redundant defs in other files, or they're just not needed anymore? maybe elaborate more in the description can help

Comment (achartier) [reply cpp/tensorrt_llm/batch_manager/trtGptModelInflightBatching.h:1]: Extended comment to clarify.

**Comment by niukuo** [comment]
[PR_Github #326](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/326/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #305](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/305/) completed with status: 'FAILURE'

**Comment by niukuo** [comment]
[PR_Github #329](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/329/) [ run ] triggered by Bot

**Comment by niukuo** [comment]
[PR_Github #329](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/329/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #309](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/309/) completed with status: 'SUCCESS'

**Review by symphonylyh — APPROVED** [review]

**Comment by niukuo** [comment]
[PR_Github #343](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/343/) [ reuse-pipeline ] triggered by Bot

**Comment by niukuo** [comment]
[PR_Github #343](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/343/) [ reuse-pipeline ] completed with state `SUCCESS`
Reusing [PR_Github #329](https://prod.blsm.nvidia.com/sw-tensorrt-top-1/job/LLM/job/helpers/job/PR_Github/329/) for commit `ee4a337`

**Comment by niukuo** [comment]
[PR_Github #358](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/358/) [ reuse-pipeline ] triggered by Bot

**Comment by niukuo** [comment]
[PR_Github #358](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/358/) [ reuse-pipeline ] completed with state `SUCCESS`
Reusing [PR_Github #329](https://prod.blsm.nvidia.com/sw-tensorrt-top-1/job/LLM/job/helpers/job/PR_Github/329/) for commit `29efde1`
